kin-sdk/kin-sdk-lib/src/androidTest/kotlin/kin/sdk/KinAccountIntegrationTest.kt

```diff
@@ -351,15 +485,72 @@ class KinAccountIntegrationTest {
         kinAccountSender.sendTransactionSync(transaction)
     }
 
+    @Test
+    @Throws(Exception::class)
+    fun accountLinkingUsingDifferentKinClients_SendTransaction_TransactionSuccess() {
+        var anotherKinClient = KinClient(InstrumentationRegistry.getTargetContext(), environment, "zxcv")
+        val masterAccount = anotherKinClient.addAccount()
+        val controlledAccount = kinClient.addAccount()
+        val destinationAccount = kinClient.addAccount()
+        onboardSingleAccount(controlledAccount, 100.0)
+        onboardSingleAccount(masterAccount, 100.0)
+        onboardSingleAccount(destinationAccount, 100.0)
+
+        linkAccount(controlledAccount, masterAccount, "some package id ")
+
+        val transaction = masterAccount.transactionBuilderSync
+                .setFee(fee)
+                .setMemo("master to destination")
+                .addOperation(PaymentOperation.Builder(
+                        KeyPair.fromAccountId(destinationAccount.publicAddress), AssetTypeNative(), "21.12300")
+                        .setSourceAccount(KeyPair.fromAccountId(controlledAccount.publicAddress))
+                        .build())
+                .build()
+        val transactionId = masterAccount.sendTransactionSync(transaction)
+
+        assertThat(transactionId.id(), not(isEmptyOrNullString()))
+        // The controlled account need to reduced the fee from the link account transaction.
+        assertThat(controlledAccount.balanceSync.value(), equalTo(BigDecimal("78.87700").subtract(feeInKin.multiply(BigDecimal(2)))))
+        assertThat(masterAccount.balanceSync.value(), equalTo(BigDecimal("100.00000").subtract(feeInKin)))
+        assertThat(destinationAccount.balanceSync.value(), equalTo(BigDecimal("121.12300")))
+        val packageIdInBase64 = masterAccount.accountDataSync.data()[controlledAccount.publicAddress]
+        assertThat(String(kin.base.codec.Base64.decodeBase64(packageIdInBase64)), containsString("some package id"))
+    }
+
+    private fun linkAccount(controlledAccount: KinAccount, masterAccount: KinAccount, managerDataValue: String) {
+        val transactionBuilder = controlledAccount.transactionBuilderSync
```

Comment:
since the master account is signing first, it makes sense that you use here the masterAccount transactionBuilder

Reply:
The master account doesn't sign first.
When we build a transaction builder we sign the transaction in the builder method.